.github/workflows/deploy.yml

```diff
@@ -39,15 +39,37 @@ jobs:
       run: |
         TAG=${GITHUB_SHA}
         echo ::set-env name=TAG::${TAG}
-        echo ::set-env name=DOCKER_BUILDKIT::1
 
-    - name: build and push docker image
-      uses: docker/build-push-action@v1.1.0
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1 
       with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
-        repository: ${{ env.DOCKERHUB_REPOSITORY }}
-        tags: ${{ env.TAG }}
+
+    - name: Build and push docker dev-build image
+      uses: docker/build-push-action@v2
+      with:
+        build-args: BUILDKIT_INLINE_CACHE=1
+        cache-from: |
+          ${{ env.DOCKERHUB_REPOSITORY }}:build-cache-${{ env.BRANCH }}
+        push: true
+        tags: ${{ env.DOCKERHUB_REPOSITORY }}:build-cache-${{ env.BRANCH }}
```

Comment:
Can we call it `dev-build` as well? It may be clearer.

Reply:
We are following the Docker pattern of name the build stage `builder`